### ISSUE TITLE
Annotations on Namespace override Service annotations

### ISSUE THREAD

**Issue by ryansimmen** [linkerd/linkerd2#6117]
### What is the issue?

Applying a `config.linkerd.io/opaque-ports` annotation on a `Namespace` will override all `Service` level annotations such as those used for `external-dns`

### How can it be reproduced?

Apply a `config.linkerd.io/opaque-ports` annotation to a `Namespace` and observe that `linkerd2` will override all `Service` level annotations such as those used for `external-dns`

**Comment by olix0r** [comment]
@ryansimmen To be clear, we're actively removing annotations unrelated to linkerd?

**Comment by ryansimmen** [comment]
> @ryansimmen To be clear, we're actively removing annotations unrelated to linkerd?

Correct

**Comment by mateiidavid** [comment]
Hey 👋🏻  thanks for bringing this up. I managed to reproduce this.

I deployed `emojivoto` to a k3d cluster. On the `emojivoto` namespace I added an opaque ports annotation:

```
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    config.linkerd.io/opaque-ports: "22"
    linkerd.io/inject: enabled
  name: emojivoto
  resourceVersion: "105722"
status:
  phase: Active
```

Next, I took a random service and I deleted it from the cluster. I added some additional annotations to its manifest (e.g `external-dns.alpha.kubernetes.io/ttl: "10"`) and then re-deployed it to the k3d cluster.

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    acme.io/foo: bar
    external-dns.alpha.kubernetes.io/internal-hostname: nginx.internal.example.org.
    external-dns.alpha.kubernetes.io/ttl: "10"
  creationTimestamp: "2021-05-13T09:14:06Z"
  name: web-svc
  namespace: emojivoto
  resourceVersion: "105676"
  uid: d2ec67aa-18e8-417f-9ae0-1a9107786201
spec:
  ...
```

After redeploying, the service only had the `opaque ports` annotation on it:
```
 k get svc web-svc -o yaml -n emojivoto
apiVersion: v1
kind: Service
metadata:
  annotations:
    config.linkerd.io/opaque-ports: "22"
  name: web-svc
  namespace: emojivoto
```

The issue comes from the way the proxy injector creates its patches to add the opaque ports annotation on a service if it's found on its namespace:
```
# proxy injector logs
time="2021-05-13T09:21:03Z" level=info msg="received service/web-svc"
time="2021-05-13T09:21:03Z" level=debug msg="using namespace emojivoto config.linkerd.io/opaque-ports annotation value"
time="2021-05-13T09:21:03Z" level=info msg="annotation patch generated for: service/web-svc"
time="2021-05-13T09:21:03Z" level=debug msg="annotation patch: [\n  {\n    \"op\": \"add\",\n    \"path\": \"/metadata/annotations\",\n    \"value\": {}\n  },\n  {\n    \"op\": \"add\",\n    \"path\": \"/metadata/annotations/config.linkerd.io~1opaque-ports\",\n    \"value\": \"22\"\n  }\n]"
```

From the logs above, we can see that the generated patch creates both the annotations map and the opaque ports annotation. We don't want to create an annotations map as part of the patch if the workload already has annotations on it; I addressed this in [PR#6119](https://github.com/linkerd/linkerd2/pull/6119).
